torch/autograd/graph.py

```diff
@@ -114,15 +115,7 @@ class save_on_cpu():
     """
     def __init__(self, pin_memory=False):
         def pack_to_cpu(tensor):
-            if not pin_memory:
-                return (tensor.device, tensor.cpu())
-
-            packed = torch.empty(
-                tensor.size(),
-                dtype=tensor.dtype,
-                layout=tensor.layout,
-                pin_memory=(torch.cuda.is_available() and not tensor.is_sparse))
-            packed.copy_(tensor)
+            packed = tensor.to("cpu", non_blocking=pin_memory)
```

Comment:
I think we want a comment block above the init that explains why it is ok to do non-blocking both ways here.

Also we should have a scary comment here and in the unpack to make sure that no one will ever try to use the "packed" CPU Tensor or modify it in any ways.